src/config.rs

```diff
@@ -147,7 +208,50 @@ pub(crate) struct InternalConfig {
     pub(crate) external_port: Option<u16>,
     pub(crate) external_ip: Option<IpAddr>,
     pub(crate) upnp_lease_duration: Duration,
-    pub(crate) min_retry_duration: Duration,
+    pub(crate) retry_config: RetryConfig,
+}
+
+/// Retry config that has passed validation.
+///
+/// Retry configurations for establishing connections and sending messages.
+/// Determines the retry behaviour of requests, by setting the back off strategy used.
+#[derive(Clone, Debug, Copy)]
+pub struct RetryConfig {
+    /// The initial retry interval.
+    pub initial_retry_interval: Duration,
+    /// The maximum value of the back off period. Once the retry interval reaches this
+    /// value it stops increasing.
+    ///
+    /// This is the longest duration we will have,
+    /// for establishing connections and sending messages.
+    /// Retrying continues even after the duration times have reached this duration.
+    /// The number of retries before that happens, will be decided by the `retry_delay_multiplier`.
+    /// The number of retries after that, will be decided by the `retrying_max_elapsed_time`.
+    pub max_retry_interval: Duration,
+    /// The value to multiply the current interval with for each retry attempt.
+    pub retry_delay_multiplier: f64,
+    /// The randomization factor to use for creating a range around the retry interval.
+    ///
+    /// A randomization factor of 0.5 results in a random period ranging between 50% below and 50%
+    /// above the retry interval.
+    pub retry_delay_rand_factor: f64,
+    /// The maximum elapsed time after instantiating
+    ///
+    /// Retrying continues until this time has elapsed.
+    /// The number of retries before that happens, will be decided by the other retry config options.
+    pub retrying_max_elapsed_time: Duration,
+}
+
+impl Default for RetryConfig {
+    fn default() -> Self {
+        Self {
+            initial_retry_interval: Duration::from_millis(200),
+            max_retry_interval: Duration::from_secs(10),
+            retry_delay_multiplier: 1.5,
+            retry_delay_rand_factor: 0.3,
+            retrying_max_elapsed_time: Duration::from_secs(30),
```

Comment:
```suggestion
            retrying_max_elapsed_time: DEFAULT_RETRYING_MAX_ELAPSED_TIME,
```
why not use these DEFAULTs?